evm/src/cpu/bootstrap_kernel.rs

```diff
@@ -64,8 +64,10 @@ pub(crate) fn eval_bootstrap_kernel<F: Field, P: PackedField<Scalar = F>>(
     let next_values: &CpuColumnsView<_> = vars.next_values.borrow();
 
     // IS_BOOTSTRAP_KERNEL must have an init value of 1, a final value of 0, and a delta in {0, -1}.
-    let local_is_bootstrap = P::ONES - local_values.op.into_iter().sum::<P>();
-    let next_is_bootstrap = P::ONES - next_values.op.into_iter().sum::<P>();
+    let local_is_bootstrap =
+        P::ONES - local_values.op.into_iter().sum::<P>() - local_values.halt_state;
+    let next_is_bootstrap =
+        P::ONES - next_values.op.into_iter().sum::<P>() - next_values.halt_state;
```

Comment:
Organization: I think it would be neater to have `halt_state` be the special, implicit flag (that's set when all other flags are 0). Intuitively, `is_bootstrap` does _something_ whereas `halt_state` does nothing. Not a dealbreaker though.